samples/nrf9160/aws_fota/README.rst

```diff
@@ -31,56 +112,133 @@ Building and running
+
+Use provisioned certificates:
+   If this option is checked, the sample uses the certificates that are stored in the modem.
+   Uncheck this option if you want to use the certificates added to the ``certificates.h`` file.
+   The sample then stores these certificates with the security tag that is defined as :ref:`Secure tag for TLS credentials <security_tag>`.
+
+   .. warning::
+      The sample will overwrite existing certificates that are stored with the configured security tag.
+
+For all other values, use the default values unless you are using a custom MQTT server.
+
+Testing
+=======
+
+After programming the sample to the board, test it by performing the following steps:
+
+1. |connect_terminal|
+#. Reset the board.
+#. Confirm that the sample prints the configured application version and connects to AWS IoT.
+   You should see output like the following::
+
+      ***** Booting Zephyr OS build v1.14.99-ncs3-snapshot2-1281-g40b430ba977c *****
+
+      The MQTT AWS Jobs FOTA Sample
+      nrf_inbuilt_key_delete(42, 0) => result=0
+      nrf_inbuilt_key_delete(42, 1) => result=0
+      nrf_inbuilt_key_delete(42, 2) => result=0
+      nrf_inbuilt_key_write => result=0
+      nrf_inbuilt_key_write => result=0
+      nrf_inbuilt_key_write => result=0
+      LTE Link Connecting ...
+      LTE Link Connected!
+      IPv4 Address 0xeeb73a0d
+      client_id: nrf-aws-fota
+      [mqtt_evt_handler:129] MQTT client connected!
+
+   (?any data in there that should not be public or that is confusing?)
+#. Log on to the `AWS IoT console`_, go to **Manage** > **Things**, and select your thing.
+#. Go to **Shadow** and confirm that the application version (``nrfcloud__dfu_v1__app_v``) is the one that you configured for the sample.
+#. In the :ref:`configuring`, change the application version.
+   Then rebuild the application, but do not program it.
+#. Go to `firmware.nrfcloud.com`_ and sign in.
+   If you already have an nRF Cloud account, you can use the same credentials.
+   Otherwise, create an account and sign in.
+#. In the menu on the left-hand side, select **Firmware**.
+#. Click **Generate new** to create a URL for the firmware image.
+#. Click **Upload** for the URL that you created and select the file ``app_update.bin`` (located in the ``zephyr`` subfolder of your build directory).
+#. Create a job document (a text file) with the following content, replacing *host_url* with the server part of the URL that you created (for example, ``s3.amazonaws.com``) and *file_path* with the path and file name (for example, ``/nordic-firmware-files/0943dfbf-cb10-4eb7-8277-a8b179eaf4ff``):
+   (?should we have an actual file here, or just state ``/nordic-firmware-files/XXXXXXX``?)
+
+   .. parsed-literal::
+      :class: highlight
+
+      {
+        "operation": "app_fw_update",
+        "fwversion": "v1.0.2",
+        "size": 181124,
+        "location": {
+          "protocol": "http:",
+          "host": "*host_url*",
+          "path": "*file_path*"
```

Comment:
**Important note**: `download_client` has changed so now we don't need to have the append `/` in file path this will have to be reflected in the documentation. As this will cause users trying to setup their own S3 bucket a bit of a headache.

Reply:
Oh I see that my comment was left a bit unclear. Yes full `filepath`requires slashes what's not need is the 1st one. 
**Not this**
```json
{
	"operation": "app_fw_update",
	"fwversion": "v1.0.2",
	"size": 181124,
	"location": {
		"protocol": "http",
		"host": "s3.amazonaws.com",
		"path": "/nordic-firmware-files/file"
	}
}
```
but this
```
{
	"operation": "app_fw_update",
	"fwversion": "v1.0.2",
	"size": 181124,
	"location": {
		"protocol": "http",
		"host": "s3.amazonaws.com",
		"path": "nordic-firmware-files/file"
	}
}
```

It is implemented in `download_client` I think it's also documented there that path does not require an initial `/`